source/core/VirtualInstrument.cpp

```diff
@@ -531,18 +531,8 @@ void ClumpParseState::ResolveActualArgument(SubString* argument, void** ppData,
     // See if actual argument is '*' meaning use the default/unwired behavior.
     if (argument->CompareCStr(tsWildCard)) {
         _actualArgumentType = FormalParameterType();
-        if (!_actualArgumentType->IsFlat()) {
-            // Define a DefaultValue type. As a DV it will never merge to another instance.
-            // Since it has no name it cannot be looked up, but it will be freed once the TADM/ExecutionContext is freed up.
-            DefaultValueType *cdt = DefaultValueType::New(_clump->TheTypeManager(), _actualArgumentType, false);
-            cdt = cdt->FinalizeDVT();
-            if (needsAddress) {
-                *ppData = (AQBlock1*)cdt->Begin(kPARead);  // * passed as a param means null
-            }
-        } else {
-            // For flat data, the call instruction logic for VIs will initialize the callee parameter
```

Comment:
So if I understand (which I may not), this change is making it so that for **native instructions** (ie primitives) when a * parameter is passed that both Flat and Non-Flat representations result in a null parameter value.

Based on this comment it sounds like for **VI calls** for both Flat and Non-Flat data a DefaultValue is created for * parameters. So it sounds like for VI calls we should always be expecting non-null parameters for both Flat and Non-Flat when * is used.

If that interpretation is correct then do we have a test that is verifying that behavior for VI calls? Just based on the comment it sounds like we would now only be creating default values for Flat types when a VI is invoked with * as the logic for Non-Flat types was just removed.

Reply:
I have added a test to exercise different SubVI call behavior with parameter types. With and without this fix, the behavior is the same. Discussed with Milan and the comments might be outdated. Did find a couple of non-showstopper issues and filed them:
https://github.com/ni/VireoSDK/issues/360
https://github.com/ni/VireoSDK/issues/359
Milan is fine to submit this now.

Reply:
If @cglzaguilar has feedback, we will address it later in a separate PR later.